test/E2ETests/Smoke.E2E.Tests/TestDataSourceExtensibilityTests.cs

```diff
@@ -19,5 +19,116 @@ public void ExecuteTestDataSourceExtensibilityTests()
                 "CustomTestDataSourceTestMethod1 (1,2,3)",
                 "CustomTestDataSourceTestMethod1 (4,5,6)");
         }
+
+        /* This test needs a reference to "Microsoft.TestPlatform.AdapterUtilities" and "Microsoft.TestPlatform.ObjectModel" NuGet packages.
+         * It's used to debug FQN changes in test discovery and test execution.
+
+        // using System;
+        // using System.Collections.Generic;
+        // using System.Collections.ObjectModel;
+        // using System.Diagnostics;
+        // using System.IO;
+        // using System.Linq;
+        // using System.Reflection;
+        // using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter;
+        // using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution;
+        // using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Extensions;
+        // using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel;
+        // using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+        // using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+        // using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
```

Comment:
nit: Remove the nested comments

Reply:
That's there just for my reference. I'll move it to another TestClass and remove the comments in later PRs.